auth/basic/middleware.go

```diff
@@ -0,0 +1,92 @@
+package basic
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/subtle"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-kit/kit/endpoint"
+	httptransport "github.com/go-kit/kit/transport/http"
+)
+
+// AuthError represents generic Authorization error
+type AuthError struct {
+	Realm string
+}
+
+// StatusCode is an implemntation of StatusCoder interface in go-kit/http
+func (AuthError) StatusCode() int {
+	return http.StatusUnauthorized
+}
+
+// Error is an implemntation of Error interface
+func (AuthError) Error() string {
+	return http.StatusText(http.StatusUnauthorized)
+}
+
+// Headers is an implemntation of Headerer interface in go-kit/http
+func (e AuthError) Headers() http.Header {
+	return http.Header{
+		"Content-Type":           []string{"text/plain; charset=utf-8"},
+		"X-Content-Type-Options": []string{"nosniff"},
+		"WWW-Authenticate":       []string{fmt.Sprintf(`Basic realm=%q`, e.Realm)}}
+}
+
+func credsAreValid(givenUser, givenPass, requiredUser, requiredPass string) bool {
+	// Equalize lengths of supplied and required credentials
+	// by hashing them
+	givenUserBytes := sha256.Sum256([]byte(givenUser))
+	givenPassBytes := sha256.Sum256([]byte(givenPass))
+	requiredUserBytes := sha256.Sum256([]byte(requiredUser))
```

Comment:
Since the required user name and password values are fixed at line 85 below, if you decide to retain this hashing technique, consider computing the hash on these two values once just before line 77, and reusing the hash instead of the raw values here.